### PR TITLE
Feat: prevent multiple claims

### PR DESCRIPTION
**Pull request by ghost**
Prevent users from claiming multiple hats

Added query for all discounts generated by user
Check if any discounts have been used before
If yes, user is ineligible to claim (also if user does not hold a valid POAP token but that logic has been added)

closes [#125](https://github.com/LinumLabs/PsyDao-FE/issues/125)

### CODE REVIEW

File: utils/determineDiscountCodeUsage.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  userHasNotUsedDiscountCode: boolean;
+};
+
+export const determineDiscountCodeUsage = async (
+  address: Address | undefined
+) => {
+  try {
+    if (!address) return;
+    const discountUsageRes = await fetch(
+      `http://localhost:3000/api/discount-usage?address=${address}`,
```
Comment (gaelicgod) [inline utils/determineDiscountCodeUsage.ts:13]: why is there a localhost here

Comment (EsterB12) [reply utils/determineDiscountCodeUsage.ts]: *facepalm* so sorry... I fixed it!

File: pages/api/discount-usage.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+          status: string;
+          usageLimit: number;
+          asyncUsageCount: number;
+        };
+      };
+    }>;
+  };
+}
+
+const SHOPIFY_API_ACCESS_TOKEN = env.SHOPIFY_API_ACCESS_TOKEN;
```
Comment (saksijas) [inline pages/api/discount-usage.ts:24]: do we have a file that contains all consts? If yes can we move this to that place?

Comment (EsterB12) [reply pages/api/discount-usage.ts:24]: not yet so I'm moving all shopify-related consts to a separate file. I will make a new PR with the rest of the consts!

File: components/shopify-widget/index.tsx [hunk 3]
```diff
@@ -28,17 +30,20 @@ const ShopifyWidget = () => {
...
+      ) {
+        setUserIsEligibleToClaim(true);
       } else {
-        setUserHoldsPoapToken(false);
+        setUserIsEligibleToClaim(false);
       }
     };
-    checkUserPOAPStatus().catch(() =>
-      console.error("Error fetching POAP status")
+    checkUserEligibilityStatus().catch(() =>
```
Comment (saksijas) [inline components/shopify-widget/index.tsx:45]: lets try to use try catch here

File: components/shopify-widget/index.tsx [hunk 4]
```diff
@@ -28,17 +30,20 @@ const ShopifyWidget = () => {
...
+    const checkUserEligibilityStatus = async () => {
+      const userEligibilityData = await handleEligibilityLogic(address);
 
-      if (userPOAPData) {
-        setUserHoldsPoapToken(true);
+      if (
+        userEligibilityData.userPOAPStatus &&
+        userEligibilityData.userHasNotUsedDiscount?.userHasNotUsedDiscountCode
+      ) {
+        setUserIsEligibleToClaim(true);
```
Comment (saksijas) [inline components/shopify-widget/index.tsx:40]:
```        const userEligibilityData: EligibilityData = await handleEligibilityLogic(address)
        const isEligible = userEligibilityData.userPOAPStatus && 
                           userEligibilityData.userHasNotUsedDiscount?.userHasNotUsedDiscountCode

        setIsEligibleToClaim(isEligible)```

**Review by saksijas — APPROVED** [review]

**Pull request merged**